Source/Redemption/Private/Bone.cpp

```diff
@@ -10,12 +12,14 @@ ABone::ABone()
 	PrimaryActorTick.bCanEverTick = true;
 
 	BoxComp = CreateDefaultSubobject<UBoxComponent>(TEXT("Box Collider"));
-	SetRootComponent(BoxComp);
 
 	BoxComp->OnComponentBeginOverlap.AddDynamic(this, &ABone::OnBoneBeginOverlap);
 
 	BoneMesh = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("Bone Mesh"));
-	BoneMesh->SetupAttachment(BoxComp);
```

Comment:
Why did you change this? Because I was mostly told to put box component (box collider) to be root component. is there any reason?

Reply:
When the bone moves what is moving is the static mesh. it is receiving the impulse. If the root is the box component it will not follow the mesh.
I changed it so the box component follows the mesh. 
But you are right in the point that in most cases it will be the box component to be the root

Reply:
Aha thank you so much^-^